modules/cloudfront_distribution/tf_module/main.tf

```diff
@@ -31,61 +30,99 @@ resource "aws_cloudfront_distribution" "s3_distribution" {
     }
   }
 
-  dynamic "custom_error_response" {
-    for_each = var.status_404_page_file == null ? [] : [1]
-    content {
-      error_caching_min_ttl = 10
-      error_code            = 404
-      response_code         = 404
-      response_page_path    = var.status_404_page_file
+  restrictions {
+    geo_restriction {
+      restriction_type = "none"
     }
   }
 
-  dynamic "custom_error_response" {
-    for_each = var.status_500_page_file == null ? [] : [1]
+  viewer_certificate {
+    cloudfront_default_certificate = var.acm_cert_arn == null ? true : false
+    acm_certificate_arn            = var.acm_cert_arn
+    ssl_support_method             = "sni-only"
+  }
+
+  dynamic "origin" {
+    for_each = var.eks_load_balancer_enabled == true ? [1] : []
     content {
-      error_caching_min_ttl = 10
-      error_code            = 500
-      response_code         = 500
-      response_page_path    = var.status_500_page_file
+      domain_name = var.load_balancer
+      origin_id   = local.lb_origin_id
+      custom_origin_config {
+        http_port              = 80
+        https_port             = 443
+        origin_protocol_policy = "http-only"
+        origin_ssl_protocols   = ["TLSv1.2", "TLSv1.1", "TLSv1"]
+      }
     }
   }
 
-  aliases = var.domains
-
-  default_cache_behavior {
-    allowed_methods  = ["GET", "HEAD", "OPTIONS"]
-    cached_methods   = ["GET", "HEAD", "OPTIONS"]
-    target_origin_id = local.s3_origin_id
-
-    forwarded_values {
-      query_string = true
-      headers      = ["Origin", "Access-Control-Request-Headers", "Access-Control-Request-Method"]
+  dynamic "origin" {
+    for_each = var.s3_load_balancer_enabled == true ? [1] : []
+    content {
+      domain_name = data.aws_s3_bucket.current_bucket[0].bucket_regional_domain_name
+      origin_id   = local.s3_origin_id
 
-      cookies {
-        forward = "none"
+      s3_origin_config {
+        origin_access_identity = var.origin_access_identity_path
       }
     }
-
-    viewer_protocol_policy = "redirect-to-https"
-    compress               = true
-    min_ttl                = 0
-    default_ttl            = 3600
-    max_ttl                = 86400
   }
 
+  dynamic "default_cache_behavior" {
+    for_each = var.eks_load_balancer_enabled == true ? [1] : []
+    content {
+      allowed_methods        = var.allowed_methods
+      cached_methods         = var.cached_methods
+      target_origin_id       = local.lb_origin_id
+      viewer_protocol_policy = "redirect-to-https"
+
+      forwarded_values {
+        query_string = true
+        headers      = ["Origin", "Access-Control-Request-Headers", "Access-Control-Request-Method"]
+
+        cookies {
+          forward = "none"
+        }
+      }
+    }
+  }
 
-  price_class = var.price_class
+  dynamic "default_cache_behavior" {
```

Comment:
duplicate block, with only difference seeming to be target_origin_id

Reply:
merged both the blocks into one.